src/main/java/com/google/sps/servlets/NewMatchQueryServlet.java

```diff
@@ -28,7 +28,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // Find the possible matches.
     MatchQuery matchQuery = new MatchQuery();
-    Collection<User> answer = matchQuery.query(matchRequest);
+    Collection<String> answer = matchQuery.query(matchRequest);
```

Comment:
Could you clarify why the matchQuery needs to return a Collection<String>? I would assume that gson.ToJson can take the Collection<User> as input as well.

Reply:
This is just how it made sense to me. Is there a different approach that would be better?

Reply:
sorry - my comment might not have been clear due to < > not showing up. My question is around why we need to change it from User to String

Reply:
Oh, that makes more sense. I changed it to String because the Json didn't seem to be working on the entire collection. I can definitely try again if you think the other way is better

Reply:
Returning a Collection of Users seems like a better solution than returning a Collection of String since it separates the string formatting concern from the logic for retrieving data.

If you return the User Collection here, the servlet response will be a json object array with the following format:

[{"id":0,"name":"Andre Harder","specialties":["DoS","Machine Learning"]},{"id":0,"name":"Julie Johnson","specialties":["Security","Machine Learning"]}]

If you add an id field to your javascript User object, the json can then be converted into an Array of your javascript User class (i.e. displayNewMatchPopup can take an Array of User instead of an Array of string).

Let me know if that makes sense.

Reply:
Yep, that makes sense. I just updated it, let me know what you think

Reply:
thanks, this looks good!